TestProject/Lesson9.cpp

```diff
@@ -0,0 +1,153 @@
+#include "Lesson7.h"
+#include "Lesson9.h"
+
+#include <iostream>
+#include <string>
+
+namespace Lesson9
+{
+	int const Task1ArraySize = 10;
+	int const Task2LineSize = 20;
+	void Lesson()
+	{
+		// #1
+		int arr[Task1ArraySize];
+		for (int i = 0; i < Task1ArraySize; i++)
+		{
+			arr[i] = ReadIntValue("[" + std::to_string(i) + "] ");
+		}
+		TranslateArray(arr, Task1ArraySize);
+		std::cout << "Translated array: " << std::endl;
+		for (int i = 0; i < Task1ArraySize; i++)
+		{
+			WriteIntAndTextValue("[" + std::to_string(i) + "] ", arr[i]);
+		}
+
+		// #2
+		char current[Task2LineSize];
+		ToUppercase(current, ReadLineValue("Input: ", current));
+		WriteCharsAndTextValue("Output: ", current);
+
+		//#3
+		WriteCharsAndTextValue("Output: ", IsPalindrom(current, ReadLineValue("Input: ", current)) ? "True" : "False");
+
+		//#4
+		int vowelsCount = 0;
+		int consonantsCount = 0;
+		ParseStringLetters(current, ReadLineValue("Input: ", current), vowelsCount, consonantsCount);
+		WriteIntAndTextValue("vowelsCount: ", vowelsCount);
+		WriteIntAndTextValue("consonantsCount: ", consonantsCount);
+
+		//#5
+		char current2[Task2LineSize];
+		WriteCharsAndTextValue("Output: ", IsEqual(current, ReadLineValue("Input1: ", current),current2, ReadLineValue("Input2: ", current2)) ? "IsEqual" : "NotEqual");
+	}
+
+	void TranslateArray(int numbers[], int size)
+	{
+		for (int i = 0; i < size; i++)
+		{
+			if (numbers[i] > 0) 
+			{
+				numbers[i] *= 2;
+			}
+			else {
+				numbers[i] = 0;
+			}
+		}
+	}
+
+	void ToUppercase(char str[], int size)
```

Comment:
you shouldn't pass size here. and shouldn't calculate it by yourself. you can use strlen(str)